### PR TITLE
Microsoft.security insights 2021 09 01 preview add missing resources

### PR DESCRIPTION
**Pull request by anat-gilenson**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [X] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [X] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit (including refactoring) and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by anat-gilenson** [comment]
Reverted Resource definition name change as it spans several PRs. Will be done in a separate PR.

**Comment by anat-gilenson** [comment]
@changlong-liu We understand the ModelValidation error is due to duplicate definitions of Resource - one in our common directory and one in the common-type directory. This validation fail is surprising as we didn't have these failures in previous PRs where systemData was used and we never reference the common-type Resource. 

The reason we have such definition in the first place is that the common-type Resource lacks the additional property systemData which ARM requires each resource to have in it's top-level properties as of 2020. (https://armwiki.azurewebsites.net/api_contracts/ResourceSystemData.html)

Is there a possibility to add such Resource definition in common-types?

**Comment by anat-gilenson** [comment]
@changlong-liu List of cross-version breaking changes and their reasons:
1. **Removed paths in bookmarks when compared to stable version:** We didn't remove any path. In the preview versions the provider is inserted as a parameter instead of being constant. This new version is aligned with previous previews.
2. **Removed definitions in Bookmarks:** The 2020 stable version is outdated. In the preview we don't have `UebaSettings` and `ToggleSettings`.
3. **Removal of `ResourceGroupName`, `SubscriptionId` and `ApiVersion` when compared to stable**: In the newer preview versions we use the common types instead of declaring duplicate definitions. We want to keep our Swagger organized.
4. **Removal of `critical` from `IncidentSeverity` enum:** this was a mistake in previous version - no such severity exists for incidents. 
5. **Comparison of our common types and the common-types types:** Indeed, we don't have all the definitions the common-types have and we shouldn't. Seems like the validation ran as both files are named the same. Should we rename our file?
6. **The new version has a different 'pattern' value than the previous one in ThreatIntelligence:** We replaced the local definitions of parameters with the common-type ones to keep our Swagger organized.

**Comment by changlong-liu** [comment]
Hi @jianyexi , would you help on @anat-gilenson 's ModelValidation questions? Thanks!

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/dataConnectors.json [hunk 1]
```diff
@@ -284,184 +284,169 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/dataConnectorsCheckRequirements": {
```
Comment (pilor) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/dataConnectors.json:288]: Please call out what APIs/definitions are new/changed in this file.  Looks like it is a combination of an existing file with APIs from another file that existed in a previous version

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/common/IncidentTypes.json [hunk 2]
```diff
@@ -0,0 +1,244 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2021-09-01-preview",
+    "title": "Common Incident types"
+  },
+  "paths": {},
+  "definitions": {
```
Comment (pilor) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/common/IncidentTypes.json:8]: Did any of these definitions get updated during the refactoring?  In the future when doing refactoring of specs do all refactoring in a few commits then all changes afterwards to make the diff clean (if that was done please point me to the commit to review from)

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/dataConnectors.json (hunk 1 above)
Comment (anat-gilenson) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/dataConnectors.json:288]: The purpose of this PR is creating a version that contains all our supported endpoints, without introducing any change.
I started this version based of 2021-03-01-preview using OpenAPI Hub and then manually added missing endpoint from 2019-01-01-preview. None of the endpoints are new, but some are missing when compared to 2019 as they aren't supported anymore. The missing endpoints:

- "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/aggregations/{aggregationsName}"
- "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/cases"
- "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/cases/{caseId}"

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/common/IncidentTypes.json (hunk 2 above)
Comment (anat-gilenson) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/common/IncidentTypes.json:8]: So sorry for the unorganized refactoring. Will make sure to do as you suggested in my next PRs.
The only change is in `IncidentInfo` which had a an enum with `Critical` value under it's `severity` property. This was a mistake in previous versions as such a severity value doesn't exist for incidents. I referenced the correct `IncidentSeverityEnum` in the severity property instead.

**Comment by anat-gilenson** [comment]
> @changlong-liu List of cross-version breaking changes and their reasons:
> 
> 1. **Removed paths in bookmarks when compared to stable version:** We didn't remove any path. In the preview versions the provider is inserted as a parameter instead of being constant. This new version is aligned with previous previews.
> 2. **Removed definitions in Bookmarks:** The 2020 stable version is outdated. In the preview we don't have `UebaSettings` and `ToggleSettings`.
> 3. **Removal of `ResourceGroupName`, `SubscriptionId` and `ApiVersion` when compared to stable**: In the newer preview versions we use the common types instead of declaring duplicate definitions. We want to keep our Swagger organized.
> 4. **Removal of `critical` from `IncidentSeverity` enum:** this was a mistake in previous version - no such severity exists for incidents.
> 5. **Comparison of our common types and the common-types types:** Indeed, we don't have all the definitions the common-types have and we shouldn't. Seems like the validation ran as both files are named the same. Should we rename our file?
> 6. **The new version has a different 'pattern' value than the previous one in ThreatIntelligence:** We replaced the local definitions of parameters with the common-type ones to keep our Swagger organized.

@pilor please note this comment which elaborates all cross-version breaking change in this PR.

**Review by changlong-liu — APPROVED** [review]

**Pull request merged**